### PR TITLE
feat: move StarkEvents to storage 

### PR DESCRIPTION
**Pull request by tonypony220**
# Pull Request type

Please add the labels corresponding to the type of changes your PR introduces:

- Feature

## What is the current behavior?

We store Starknet events in Substrate events which is not pretty convenient and infate [block size](https://github.com/paritytech/substrate/blob/033d4e86cc7eff0066cd376b9375f815761d653c/frame/system/src/lib.rs#L627)

Resolves: #NA

## What is the new behavior?

- `TxEvents` are not deleted after storing block, so that it's main storage for starknet events
- `StarkEvent`s does not stored in Substrate events

## Does this introduce a breaking change?

No

### CODE REVIEW

**Comment by tonypony220** [comment]
Should we store in [Box](https://github.com/paritytech/substrate/blob/033d4e86cc7eff0066cd376b9375f815761d653c/frame/system/src/lib.rs#L633) as well?

**Comment by tdelabro** [comment]
> Should we store in [Box](https://github.com/paritytech/substrate/blob/033d4e86cc7eff0066cd376b9375f815761d653c/frame/system/src/lib.rs#L633) as well?

I don't think so

File: crates/pallets/starknet/src/lib.rs [hunk 1]
```diff
@@ -409,11 +409,9 @@ pub mod pallet {
     /// EVENTS
     /// See: `<https://docs.substrate.io/main-docs/build/events-errors/>`
     #[pallet::event]
```
Comment (tdelabro) [inline crates/pallets/starknet/src/lib.rs:411]: Can you remove the whole Event section? If it is not used at all

File: crates/pallets/starknet/src/lib.rs [hunk 2]
```diff
@@ -988,7 +986,7 @@ impl<T: Config> Pallet<T> {
         BlockHash::<T>::insert(block_number, blockhash);
 
         // Kill pending storage.
-        // There is no need to kill `TxEvents` as we used `take` while iterating over it.
+        // There is no need to kill `TxEvents` as we store them.
```
Comment (tdelabro) [inline crates/pallets/starknet/src/lib.rs:989]: Remove the whole comment :)

File: crates/pallets/starknet/src/lib.rs (hunk 1 above)
Comment (tonypony220) [reply crates/pallets/starknet/src/lib.rs]: done

**Comment by tdelabro** [comment]
It's weird. The CI pass when I run it in local!

**Pull request merged**